src/app/containers/SocialEmbed/sourceHelpers.js

```diff
@@ -0,0 +1,30 @@
+const PROVIDERS = {
+  TWITTER: 'twitter',
+  UNKNOWN: 'unknown',
+};
+
+/**
+ * Determine the name of a supported provider from a social embed source URL.
+ * @param {string} source The source URL of a social embed.
+ * @returns {string} Provider name.
+ */
+export const getProviderFromSource = source => {
+  if (source.match(/^https:\/\/twitter\.com/)) {
+    return PROVIDERS.TWITTER;
+  }
+  return PROVIDERS.UNKNOWN;
+};
+
+/**
+ * Extract the social media embed ID from a supported provider's social embed
+ * source URL.
+ * @param {string} source The source URL of a social embed.
+ * @returns {string} Social embed ID.
+ */
+export const getIdFromSource = source => {
+  const provider = getProviderFromSource(source);
+  if (provider === PROVIDERS.TWITTER) {
+    return source.match(/\/status\/([0-9]+)/)[1];
```

Comment:
Any risk of an uncaught exception here?

![Screen Shot 2021-04-22 at 13 08 05](https://user-images.githubusercontent.com/8601460/115711728-ccef6000-a36b-11eb-8956-20156b896c33.png)

Reply:
Even though `source` should never be undefined, it would be better to write this defensively. I will tweak it - thanks, Ryan!

Reply:
👍 Don't think the problem is source potentially being undefined - rather source potentially not matching the regex

Therefore, `match` would return null, and then trying to index into a null reference will be an uncaught exception

Reply:
That was a good spot - we no longer return on no match. I've also added a test case to cover this scenario.